prometheus/src/main/java/com/alibaba/nacos/prometheus/controller/PrometheusController.java

```diff
@@ -73,26 +72,73 @@ public ResponseEntity metric() throws NacosException {
                 List<? extends Instance> instances = instanceServiceV2.listAllInstances(namespace,
                         service.getGroupedServiceName());
                 
-                for (Instance instance : instances) {
-                    targetSet.add(instance);
-                }
+                targetSet.addAll(instances);
                 
             }
         }
-        Map<String, List<Instance>> groupingInsMap = targetSet.stream().collect(groupingBy(Instance::getClusterName));
-        groupingInsMap.forEach((key, value) -> {
-            ObjectNode jsonNode = JacksonUtils.createEmptyJsonNode();
-            ArrayNode targetsNode = JacksonUtils.createEmptyArrayNode();
-            ObjectNode labelNode = JacksonUtils.createEmptyJsonNode();
-            value.forEach(e -> {
-                targetsNode.add(e.getIp() + ":" + e.getPort());
-            });
-            labelNode.put("__meta_clusterName", key);
-            jsonNode.replace("targets", targetsNode);
-            jsonNode.replace("labels", labelNode);
-            arrayNode.add(jsonNode);
+        
+        PrometheusUtils.assembleArrayNodes(targetSet, arrayNode);
+        return ResponseEntity.ok().body(arrayNode.toString());
+    }
+    
+    
+    /**
+     * Get service instances from designated namespace.
+     *
+     * @throws NacosException NacosException.
+     */
+    @GetMapping(value = ApiConstants.PROMETHEUS_CONTROLLER_NAMESPACE_PATH, produces = "application/json; charset=UTF-8")
+    public ResponseEntity metricNamespace(@PathVariable("namespace") String namespace) throws NacosException {
```

Comment:
Should we use namespaceId?  nacos has namespaceShowName and namespaceId, directly use namespace may be let users wrong input namespaceShowName.

Reply:
I thought "namespace" is "namespaceId"..  Isn't it?   If it is, may be variable name in Service.class should be changed, If not, I can try find a way to get namespaceId....

Reply:
Yes it's, but I mean users may be confused and may use `showName`

Reply:
Only change the parameters name

Reply:
> Should we use namespaceId?  nacos has namespaceShowName and namespaceId, directly use namespace may be let users wrong input namespaceShowName.